src/clipboard.rs

```diff
@@ -68,4 +68,27 @@ impl Clipboard {
         }
         success
     }
+
+    pub fn request_uris<P: FnOnce(&Clipboard, &[glib::GString]) + 'static>(&self, callback: P) {
+        let callback_data: Box_<P> = Box_::new(callback);
+        unsafe extern "C" fn callback_func<P: FnOnce(&Clipboard, &[glib::GString]) + 'static>(
+            clipboard: *mut gtk_sys::GtkClipboard,
+            uris: *mut *mut c_char,
+            data: glib_sys::gpointer,
+        ) {
+            let clipboard = from_glib_borrow(clipboard);
+            let uris = from_glib_borrow(uris);
```

Comment:
Just want to be sure here: does `glib_from_borrow` handle correctly pointers of pointers and stop at the first `0`/`NULL`?

Maybe you know @sdroege ?

Reply:
I don't think this compiles

Reply:
See this for an example

```rust
    pub fn get_uris(&self) -> Vec<GString> {
        unsafe {
            FromGlibPtrContainer::from_glib_full(gtk_sys::gtk_selection_data_get_uris(
                self.to_glib_none().0,
            ))
        }
    }
```

Reply:
Should this be `from_glib_full`, `container` or `none`? The docs have no transfer annotation.

Reply:
`none`

Reply:
Okay, take two.